File: code_samples/scala/web_development_routing.scala
/**
 * Web Development: Routing
 * AI/ML Training Sample
 */

class Routing {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object RoutingExample {
  def main(args: Array[String]): Unit = {
    val instance = new Routing()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/haskell/web_development_routing.hs
-- Web Development: Routing
-- AI/ML Training Sample

module Routing where

data Routing = Routing {
    getData :: String
} deriving (Show, Eq)

process :: Routing -> String -> Routing
process obj input = obj { getData = input }

validate :: Routing -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Routing { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
